speed-dreams/source-2.2.3/src/modules/graphic/ssggraph/grboard.cpp

```diff
@@ -1597,33 +1597,48 @@ void cGrBoard::grDispIntervention()
     glBindTexture(GL_TEXTURE_2D, 0);
+
+    tTextData textData = InterventionConfig::GetInstance()->GetCurrentInterventionText();
+    GfuiDrawString(textData.Text, normal_color_, GFUI_FONT_LARGE_C, textData.Position.X, textData.Position.Y);
+
     glPopMatrix();
 }
 
-/// @brief Loads the intervention textures from XML into the InterventionConfig singleton class.
+/// @brief Loads the intervention textures and texts from XML into the InterventionConfig singleton class.
 ///        Requires that 'data/intervention' has been added to the search filepath grFilePath.
-void LoadInterventionTextures()
+void LoadInterventionData()
 {
-    InterventionConfig* instance = InterventionConfig::GetInstance();
-    tTextureData* textures = new TextureData[instance->GetInterventionCount()];
+    InterventionConfig* config = InterventionConfig::GetInstance();
+    tTextureData* textures = new TextureData[config->GetInterventionCount()];
+    tTextData* texts = new TextData[config->GetInterventionCount()];
 
     char path[256];
-    void* xmlHandle = InterventionConfig::GetInstance()->GetXmlHandle();
-    for (int i = 0; i < instance->GetInterventionCount(); i++)
+    void* xmlHandle = config->GetXmlHandle();
+    for (int i = 0; i < config->GetInterventionCount(); i++)
     {
-        snprintf(path, sizeof(path), "%s/%d", PRM_SECT_INTERVENTIONS, i);
-        const char* name = GfParmGetStr(xmlHandle, path, PRM_ATTR_NAME, "");
-        const char* tex = GfParmGetStr(xmlHandle, path, PRM_ATTR_TEXTURE, "");
-        int xPos = GfParmGetNum(xmlHandle, path, PRM_ATTR_XPOS, NULL, 0);
-        int yPos = GfParmGetNum(xmlHandle, path, PRM_ATTR_YPOS, NULL, 0);
+        int xPos, yPos;
+
+        // Textures
+        snprintf(path, sizeof(path), "%s/%s/%s", PRM_SECT_INTERVENTIONS, s_actionEnumParamMap[i], PRM_SECT_TEXTURE);
+        const char* source = GfParmGetStr(xmlHandle, path, PRM_ATTR_SRC, "");
+        xPos = GfParmGetNum(xmlHandle, path, PRM_ATTR_XPOS, NULL, 0);
+        yPos = GfParmGetNum(xmlHandle, path, PRM_ATTR_YPOS, NULL, 0);
 
         // IMPORTANT: The texture should not be bigger than 256x256 due to buffer sizes.
-        ssgSimpleState* texture = (ssgSimpleState*)grSsgLoadTexState(tex);
+        ssgSimpleState* texture = (ssgSimpleState*)grSsgLoadTexState(source);
+        textures[i] = TextureData(texture, { xPos, yPos });
+
+        // Texts
+        snprintf(path, sizeof(path), "%s/%s/%s", PRM_SECT_INTERVENTIONS, s_actionEnumParamMap[i], PRM_SECT_TEXT);
+        const char* txt = GfParmGetStr(xmlHandle, path, PRM_ATTR_TXT, "");
+        xPos = GfParmGetNum(xmlHandle, path, PRM_ATTR_XPOS, NULL, 0);
+        yPos = GfParmGetNum(xmlHandle, path, PRM_ATTR_YPOS, NULL, 0);
 
-        textures[i] = TextureData(name, texture, { xPos, yPos });
+        texts[i] = { txt, { xPos, yPos } };
     }
 
-    InterventionConfig::GetInstance()->SetTextures(textures);
+    config->SetTextures(textures);
+    config->SetTexts(texts);
```

Comment:
Any reason you load in the text in here instead of on Initialze in the config class?

Reply:
Personally I think it's a little strange that it gets loaded in grboard and then saved to the config instead of just loading in the config. It could maybe be merged with the sound loop?